Improve the Firebase Account Module

PSchmiedmayer opened the pull request:
# Improve the Firebase Account Module

## :recycle: Current situation & Problem
- Firebase includes some built-in password rules that we do not expose in the API
- We do not display localized error messages for Firebase errors
- The localization of the sign-up button is not correctly set
- The application does not recognize if, e.g., a user is deleted in the Firebase Emulator or on Firebase
- The application does not refresh its state when the user is logged in

Thank you to @jenlmoore for providing valuable feedback to the CardinalKit project!

## :bulb: Proposed solution
- Includes a password rule in the UI (at least eight characters, no whitespaces)
- Adds localized descriptions for Firebase Errors
- Improves the logic in the account module to use the correct localization
- Automatically sets the user to nil and the account to be signed out if a user refresh token request did not succeed
- Adds further logic to refresh the application if a user is signed in

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

vishnuravi reviewed (approved):
Thank you @PSchmiedmayer for implementing these improvements!

vishnuravi commented:
The failed check appears to be due to an error in a clean up step. The tests are all passing.

PSchmiedmayer commented:
The clean up step failed due to the build step failing before that. I have fixed that failing dependency. 

The build error failed due to wrong key in the localization files, I have fixed this 👍
